pkg/controllers/securitypolicy/securitypolicy_controller.go

```diff
@@ -100,22 +109,36 @@ func (r *SecurityPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ResultNormal, client.IgnoreNotFound(err)
 	}
 
+	isZero := false
+	finalizerName := servicecommon.OldSecurityPolicyFinalizerName
+	realObj := &v1alpha1.SecurityPolicy{}
+	switch obj.(type) {
```

Comment:
we already make sure obj type is either &crdv1alpha1.SecurityPolicy{} when VPC enabled or 
&v1alpha1.SecurityPolicy{} for T1.
Why we still do version conversion here. It looks like we assume v1alpha1.SecurityPolicy can be exixst in VPC mode, but this is impossible. Since VPC is greenfield, there is no existing CR with crdv1alpha1.SecurityPolicy. So, in VPC mode, we can always check &crdv1alpha1.SecurityPolicy{}
and when VPC disabled, we only check T1 &v1alpha1.SecurityPolicy{}.
One thing is if we also install T1 v1alpha1.SecurityPolicy CRD in VPC mode. I think the answer is no. 
For T1, we only install v1alpha1.SecurityPolicy CRD  and install crdv1alpha1.SecurityPolicy in VPC.